src/firewall/core/fw_ifcfg.py

```diff
@@ -30,20 +30,26 @@
 from firewall.core.io.ifcfg import ifcfg
 
 def search_ifcfg_of_interface(interface):
-    """search ifcfg file for the interface in config.IFCFGDIR"""
+    """search ifcfg file for the interface in IFCFGDIR_REDHAT or IFCFGDIR_SUSE"""
 
     # Return quickly if config.IFCFGDIR does not exist
-    if not os.path.exists(config.IFCFGDIR):
+    if not os.path.exists(config.IFCFGDIR_REDHAT) and \
+       not os.path.exists(config.IFCFGDIR_SUSE):
         return None
 
-    filename = "%s/ifcfg-%s" % (config.IFCFGDIR, interface)
+    ifcfgdir, suseifcfg = (config.IFCFGDIR_REDHAT, False) if os.path.exists(config.IFCFGDIR_REDHAT) else (config.IFCFGDIR_SUSE, True)
+    filename = "%s/ifcfg-%s" % (ifcfgdir, interface)
     if os.path.exists(filename):
         ifcfg_file = ifcfg(filename)
         ifcfg_file.read()
-        if ifcfg_file.get("DEVICE") == interface:
+        # on SUSE, ETHERDEVICE is only applicable when creating VLANs so it may not always
+        # be there.
+        if ifcfg_file.get("DEVICE") == interface or \
+           ifcfg_file.get("ETHERDEVICE") == interface or \
+           (suseifcfg and ifcfg_file.get("ETHERDEVICE") is None):
```

Comment:
If SUSE does not use DEVICE or ETHERDEVICE for a regular non-VLAN, then how do we know which ifcfg to use? This looks like it will select the first file it encounters due to 
`   (suseifcfg and ifcfg_file.get("ETHERDEVICE") is None):`

Reply:
@erig0 on SUSE, the ifcfg files are named like 'ifcfg-$interface' so if $interface matches the interface we are interested in, then we use that file. Does it make sense?